File: elixir/7ku/the_fusc_function_part_1.ex
defmodule Kata do
  def fusc(0), do: 0
  def fusc(1), do: 1
  def fusc(n) when rem(n, 2) == 0, do: fusc(floor(n / 2))
  def fusc(n), do: fusc(floor((n - 1) / 2)) + fusc(floor((n + 1) / 2))
end

File: ruby/7ku/The fusc function -- Part 1.rb
# frozen_string_literal: true
# The fusc function -- Part 1
# https://www.codewars.com/kata/570409d3d80ec699af001bf9/train/ruby

def fusc(n)
  return 0 if n == 0
  return 1 if n == 1
  return fusc(n / 2) if n % 2 == 0

  fusc((n - 1) / 2) + fusc((n + 1) / 2)
end